src/game_interpreter.cpp

```diff
@@ -4643,6 +4645,34 @@ bool Game_Interpreter::CommandManiacCallCommand(lcf::rpg::EventCommand const&) {
 	return true;
 }
 
+bool Game_Interpreter::CommandGetTime(lcf::rpg::EventCommand const& com) {
+
+	std::string periodName = Utils::LowerCase(ToString(com.string));
+	int32_t outputVariable = ValueOrVariable(com.parameters[0], com.parameters[1]);
+
+	lcf::rpg::EventCommand outputCommand;
+	outputCommand.code = int(Cmd::ControlVars);
+
+	std::time_t t = std::time(nullptr);
+	std::tm* tm = std::localtime(&t);
+
+	int32_t dateOutput{};
+
+	if (periodName == "getyear") dateOutput = tm->tm_year + 1900;
+	if (periodName == "getmonth") dateOutput = tm->tm_mon + 1;
+	if (periodName == "getday") dateOutput = tm->tm_mday;
+	if (periodName == "gethour") dateOutput = tm->tm_hour;
+	if (periodName == "getminute") dateOutput = tm->tm_min;
+	if (periodName == "getsecond") dateOutput = tm->tm_sec;
+	if (periodName == "getweekday") dateOutput = tm->tm_wday +1;
+	if (periodName == "getyearday") dateOutput = tm->tm_yday +1;
+
+	std::vector<int32_t> updateVarParams = { 0, outputVariable, 0, 0, 0, dateOutput };
+	outputCommand.parameters = lcf::DBArray<int32_t>(updateVarParams.begin(), updateVarParams.end());
+
+	return ExecuteCommand(outputCommand);
```

Comment:
Is there a reason why you delegate this to the ControlVariables command instead of setting the variables directly?

Reply:
I thought it would be better to not rewrite existing code. 
and that would also cover setting up savefiles.

Though I could simply use `CommandControlVariables(outputCommand);` instead of `executeCommand(outputCommand);`

Reply:
fixed using

```cpp
Main_Data::game_variables->Set(outputVariable, dateOutput);
Game_Map::SetNeedRefresh(true);
```